src/components/smart_objects/include/smart_objects/default_shema_item.h

```diff
@@ -98,9 +99,20 @@ CDefaultSchemaItem<Type>::CDefaultSchemaItem(const ParameterType& DefaultValue)
     : mDefaultValue(DefaultValue) {}
 
 template <typename Type>
-Errors::eType CDefaultSchemaItem<Type>::validate(const SmartObject& Object) {
-  return (getSmartType() == Object.getType()) ? Errors::OK
-                                              : Errors::INVALID_VALUE;
+Errors::eType CDefaultSchemaItem<Type>::validate(const SmartObject& Object,
+                                                 std::string& errorMessage) {
+  if (getSmartType() != Object.getType()) {
+    if (!Object.getKey().empty()) {
+      errorMessage.assign("Validation failed for \"" + Object.getKey() +
```

Comment:
@jacobkeeler not sure that initial error message should be replaced. Or is in output parameter?